github/Requester.py

```diff
@@ -115,13 +120,18 @@ def close(self):
 
 class HTTPRequestsConnectionClass(object):
     # mimic the httplib connection object
-    def __init__(self, host, port=None, strict=False, timeout=None, **kwargs):
+    def __init__(self, host, port=None, strict=False, timeout=None, retry=None, **kwargs):
         self.port = port if port else 80
         self.host = host
         self.protocol = "http"
         self.timeout = timeout
         self.verify = kwargs.get("verify", True)
         self.session = requests.Session()
+        # Code to support retries
+        if retry != None:
```

Comment:
Instead of __if retry != None:__, it might be safer to use __if retry:__ to deal with the case where the if the caller passes in __retry=False__.